### PR TITLE
[CHANGELOG] Prepare v4.0.0-alpha-24

### PR DESCRIPTION
**Pull request by m4theushw**

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -3,6 +3,46 @@
...
+## [4.0.0-alpha.24](https://github.com/mui-org/material-ui-x/compare/v4.0.0-alpha.23...v4.0.0-alpha.24)
+
+_Apr 2, 2021_
+
+Big thanks to the 9 contributors who made this release possible. Here are some highlights ✨:
+
+- 🇬🇷 Added elGR locale (#1275) @clytras
+- 🇪🇸 Added esES locale (#1286) @WiXSL
+- 🇯🇵 Added jaJP locale (#1283) @seed-of-apricot
+- 🇳🇱 Added nlNL locale (#1273) @wimdetroyer
```
Comment (oliviertassinari) [inline CHANGELOG.md:15]: To follow https://www.notion.so/GitHub-issues-PRs-c1d7072e0c2545b0beb43b115f6030f6 `Pull request title & descriptions`

```suggestion
- 🇬🇷 Add elGR locale (#1275) @clytras
- 🇪🇸 Add esES locale (#1286) @WiXSL
- 🇯🇵 Add jaJP locale (#1283) @seed-of-apricot
- 🇳🇱 Add nlNL locale (#1273) @wimdetroyer
```

File: CHANGELOG.md [hunk 2]
```diff
@@ -3,6 +3,46 @@
...
+- [DataGrid] The `renderHeader` will be called with a `GridColumnHeaderParams` instead of `GridColParams` (#1299) @m4theushw
+- [DataGrid] The `apiRef.current.moveColumn` was renamed to `apiRef.current.setColumnIndex` (#1299) @m4theushw
+
+#### Changes
+
+- [DataGrid] Fix loader flag from useDemoData hook (#1279) @DanailH
+- [DataGrid] Fix page shift after toggling column (#1284) @m4theushw
+- [DataGrid] Fix rendering issues (#1319, #1253) @dtassone
+- [DataGrid] Refactor edit events to allow stop propagation (#1304) @dtassone
+- [Security] Bump y18n from 4.0.0 to 4.0.1 (#1318) @dependabot-preview
```
Comment (oliviertassinari) [inline CHANGELOG.md:40]: Noise

```suggestion
```

File: CHANGELOG.md [hunk 3]
```diff
@@ -3,6 +3,46 @@
...
+### @material-ui/x-grid@v4.0.0-alpha.24 / @material-ui/data-grid@v4.0.0-alpha.24
+
+#### Breaking Changes
+
+- [DataGrid] All slot components no longer get access to `GridBaseComponentProps` through the props. To use the `GridBaseComponentProps` call the `useGridSlotComponentProps` hook. (#1252) @DanailH
+- [DataGrid] Type `GridSlotsComponent` changed (#1252) @DanailH
+- [DataGrid] Rename `GridBaseComponentProps` type to `GridSlotComponentProps` (#1252) @DanailH
+- [DataGrid] Rename `useGridBaseComponentProps` hook to `useGridSlotComponentProps` (#1252) @DanailH
+- [DataGrid] Rename modules (#1292) @DanailH
+- [DataGrid] Renamed all events related to column reordering, e.g. `GRID_COL_REORDER_START` -> `GRID_COLUMN_REORDER_START` (#1299) @m4theushw
```
Comment (m4theushw) [inline CHANGELOG.md:27]: Not sure if this should be mentioned in the CHANGELOG. In https://github.com/mui-org/material-ui-x/pull/1250 I proposed to rename them again.

**Review by oliviertassinari — COMMENTED** [review]
Almost good

File: CHANGELOG.md (hunk 3 above)
Comment (oliviertassinari) [reply CHANGELOG.md:27]: No preferences

**Review by oliviertassinari — APPROVED** [review]

**Pull request merged**